ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/Vectorizer.java

```
@@ -2411,6 +2415,21 @@ private boolean getOnlyStructObjectInspectors(ReduceWork reduceWork,
     }
   }
 
+  private void addCustomUDFs(HiveConf hiveConf) {
+    String[] udfs =
+        HiveConf.getTrimmedStringsVar(hiveConf, HiveConf.ConfVars.HIVE_VECTOR_ADAPTOR_CUSTOM_UDF_WHITELIST);
+    if (udfs != null) {
+      for (String udf : udfs) {
+        try {
+          supportedGenericUDFs.add(Class.forName(udf));
+          LOG.info("Registered custom UDF: {}", udf);
+        } catch (ClassNotFoundException e) {
+          LOG.warn("Additional UDF not found: {}", udf);
```

please don't swallow the exception:
```LOG.warn("Failed to register custom UDF: {}", udf, e)````

---

I fixed.